conn.go

```diff
@@ -315,6 +298,84 @@ func (s *Session) dialWithoutObserver(ctx context.Context, host *HostInfo, cfg *
 	return c, nil
 }
 
+func (s *Session) dialWithEndpoint(ctx context.Context, endpoint *Endpoint, dialer Dialer) (net.Conn, error) {
+	if len(endpoint.ProxyURL) != 0 {
+		u, err := url.Parse(endpoint.ProxyURL)
+		if err != nil {
+			return nil, fmt.Errorf("can't parse proxy URL %q: %w", endpoint.ProxyURL, err)
+		}
+
+		d, err := proxy.FromURL(u, proxyDialerFunc(func(network, addr string) (net.Conn, error) {
+			return dialer.DialContext(ctx, network, addr)
+		}))
+		if err != nil {
+			return nil, fmt.Errorf("can't create proxy dialer: %w", err)
+		}
+
+		dialer = d.(proxy.ContextDialer)
+	}
+
+	var err error
+	if gocqlDebug {
+		s.logger.Println("Dialing to", endpoint.Server)
+	}
+
+	conn, err := dialer.DialContext(ctx, "tcp", endpoint.Server)
+	if err != nil {
+		return nil, fmt.Errorf("can't connect to %q: %w", endpoint.Server, err)
+	}
+	if endpoint.TLSConfig != nil {
+		if gocqlDebug {
+			s.logger.Printf("TLS Handshake using %q SNI with %q\n", endpoint.TLSConfig.ServerName, endpoint.Server)
+		}
+		tconn := tls.Client(conn, endpoint.TLSConfig)
+		if err := tconn.HandshakeContext(ctx); err != nil {
```

Comment:
Note to people using Fedora 34's provided Go: `HandshakeContext` was introduced by Go 1.17, but the latest version of Go on Fedora 34 is 1.16.15 - so you need to install Go yourself.

Reply:
Go 1.16 is no longer supported because there's 1.18 already. Leaving as is.